webapp/channels/src/types/store/draft.ts

```diff
@@ -29,13 +29,14 @@ export type PostDraft = {
             requested_ack?: boolean;
             persistent_notifications?: boolean;
         };
+        files?: FileInfo[];
     };
 };
 
 export function isPostDraftEmpty(draft: PostDraft): boolean {
     const hasMessage = draft.message.trim() !== '';
-    const hasAttachment = draft.fileInfos.length > 0 || draft.file_ids?.length;
-    const hasUploadingFiles = draft.uploadsInProgress.length > 0;
+    const hasAttachment = draft.fileInfos?.length > 0;
+    const hasUploadingFiles = draft.uploadsInProgress?.length > 0;
```

Comment:
Typescript should be shouting here. Either `fileInfos` is (by type definition) never undefined, and you made this change because you found it undefined at some point (and the type definition should be updated), or the question marks are not needed.

This probably work because `(undefined > 0) === false` but still, it is weird.

Reply:
Type definition should be updated but thats a long task. I started with it but there were too many changes involved to bundle it with another change.